sample_XX_headless/src/main.hpp

```diff
@@ -0,0 +1,165 @@
+ *
+ *  SPDX-License-Identifier: MIT
+ *
+ *  Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
+ *
+ *  1. Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+ *
+ *  2. Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the
+ *     documentation and/or other materials provided with the distribution.
+ *
+ *  3. Neither the name of the copyright holder nor the names of its contributors may be used to endorse or promote products derived from this
+ *     software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+ *  THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS
+ *  BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE
+ *  GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH
+ *  DAMAGE.
+ *
+ */
+
+
+// Generated by cmake by populating project_config.h.in
+#include "project_config.h"
+
+#include <cstdlib>
+#include <iomanip>
+#include <iostream>
+#include <future>
+#include <chrono>
+#include <thread>
+
+#include <openxr_provider.h>
+
+#ifdef LINUX
+	#include <unistd.h>
+#endif
+#ifdef WINDOWS
+	#include <windows.h>
+#endif
+
+#define APP_NAME "sample_xx_headless"
+#define ENGINE_NAME "openxr_provider"
+
+#define LOG_CATEGORY_HEADLESS "OpenXRProviderHeadless"
+
+// global vars
+XrEventDataBaseHeader *g_xrEventDataBaseheader = nullptr;
+
+// Pointer to input handling object of the openxr provider library
+oxr::Input *g_pInput = nullptr;
+
+// Current openxr session state
+XrSessionState g_sessionState = XR_SESSION_STATE_UNKNOWN;
+
+// Vive Tracker Extension, if present
+oxr::ExtHTCXViveTrackerInteraction *g_extHTCXViveTrackerInteraction = nullptr;
+
+// Future for input thread
+std::future< XrResult > g_inputThread;
+
+struct
+{
+	bool bIsActive = false;
+} g_ActionLeftElbow, g_ActionLeftFoot;
+
+struct ControllerActionData
+{
+	bool bAimTouch = false;
+	bool bAimClick = false;
+	bool bSelect = false;
+	bool bActivate = false;
+	bool bFullSpecial = false;
+	bool bIsActive = false;
+	float fSpecial = 0.0f;
+	XrVector2f xrvec2Aim = { 0.0f, 0.0f };
+} g_ActionLeftController, g_ActionRightController;
+
+inline void UpdateLeftElbowPose( oxr::Action *pAction, uint32_t unActionStateIndex )
+{
```

Comment:
Just in case anyone ever sees this, these action handlers were just cut and paste specials and I never really thought about what they should be.